### ISSUE TITLE
Dynamic Recipes

### ISSUE THREAD

**Issue by cho0pera** [StardewModders/mod-ideas#1104]
No idea if it's possible and it's likely a C# mod, but it'd be so cool to be able to have recipes that allow for ingredient substitutions. 

For example, a dish that calls for Salmon _or_ Tuna and will accept either ingredient. Or for modded JA objects, a recipe that will accept one of the listed options for carrots (there are so many versions of carrots btwn mods). 

I think it would be a framework mod possibly that mod authors can use to write new "dynamic recipes" allowing for more flexibility in cooking. 

I've seen mods that are sort of similar to this idea (Advanced Cooking and Okay Recipes) but they don't have an option for recipe ingredient substitutions. 

Using categories for ingredients is the closest thing we have now and that's neat but they're pretty broad. Having subcategories for things like vegan products within the larger -5/-6 egg/milk categories would be cool! 

Also, the fish category makes sashimi kind of annoying as any fish will become the same generic sashimi and the price values of the fish are not accounted for. (Tiered sashimi was another idea I had to make low/mid/high grade sashimi and maki rolls).

**Comment by spacechase0** [comment]
DGA allows you to use context tags for recipe ingredients - 1.6 will have this feature in SpaceCore

**Comment by cho0pera** [comment]
@spacechase0 - oh, but DGA recipe and Love of Cooking are not cooperative AFAIK (?) 
I don't know the how or why but I was told that LoC breaks DGA recipes. Is this correct?

**Comment by spacechase0** [comment]
Yeah, Love of Cooking would need to add support for SpaceCore's crafting recipe API.

**Comment by tiakall** [comment]
1.6 and Spacecore allow item context tag recipes and several mods like WAG and Cornucopia have made use of this. Not sure if LoC is compatible yet but that would be a suggestion to blueberry. :)
